Stripe/STPTheme.m

```diff
@@ -159,11 +157,29 @@ - (UIColor *)errorColor {
 }
 
 - (UIFont *)font {
-    return _font ?: STPThemeDefaultFont;
+    if (_font != nil) {
+        return _font;
+    } else {
+        if (@available(iOS 11.0, *)) {
+            UIFontMetrics *fontMetrics = [UIFontMetrics metricsForTextStyle:UIFontTextStyleBody];
+            return [fontMetrics scaledFontForFont:STPThemeDefaultFont];
+        } else {
+            return STPThemeDefaultFont;
+        }
+    }
 }
 
 - (UIFont *)emphasisFont {
-    return _emphasisFont ?: STPThemeDefaultMediumFont;
+    if (_emphasisFont != nil) {
+        return _emphasisFont;
+    } else {
+        if (@available(iOS 11.0, *)) {
+            UIFontMetrics *fontMetrics = [UIFontMetrics metricsForTextStyle:UIFontTextStyleBody];
+            return [fontMetrics scaledFontForFont:STPThemeDefaultMediumFont];
```

Comment:
What happens if we also called this for `_emphasisFont` e.g. remove the if statement above and replace this line with `scaledFontForFont:_emphasisFont ?: STPThemeDefaultMediumFont`?

More broadly I'm wondering if we could support dynamic type even for user-supplied fonts.

Reply:
yeah that _might_ be something we want to do, but I'm not sure. It seems simpler to not mess with custom set fonts to me so I was thinking have that as an initial solution and see what the response is like?